File: 2553. Separate the Digits in an Array/digit.py
class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        res = []
        
        def get_digit(number, n):
            return number // 10**n % 10
        
        for i in range(len(nums)):
            for j in range(len(str(nums[i]))):
                res.append(get_digit(nums[i], len(str(nums[i]))-j-1))
            
        return res

File: 2553. Separate the Digits in an Array/digit.cpp
#include <cmath>
class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < to_string(nums[i]).size(); j++) {
                res.push_back(fmod(nums[i] / pow(10, to_string(nums[i]).size() - j - 1), 10));
            }
        }
        return res;
    }
};